LoopKit/CarbKit/CarbStore.swift

```diff
@@ -252,7 +252,11 @@ public final class CarbStore: HealthKitSampleStore {
             cacheStore.fetchAnchor(key: CarbStore.healthKitQueryAnchorMetadataKey) { (anchor) in
                 self.queue.async {
                     self.queryAnchor = anchor
-            
+
+                    if !self.authorizationRequired {
```

Comment:
@ps2 Maybe I'm missing something, but `!authorizationRequired` doesn't mean "authorization is not needed", but actually "authorization _is_ needed [but hasn't been granted yet]".  So, can you explain please why this (and all the others) are okay?  Thanks.

Reply:
Right. `!authorizationRequired` means "we've already asked the user for auth". In those cases where we've already asked, we can go ahead and start the query right here. If we haven't asked, we assume that eventually Loop will trigger the ask (`healthStore.requestAuthorization`), and then call `authorize()` on the stores, which will trigger the query creation.

Reply:
Ah, ok, I think I see now.